### ISSUE TITLE
Deprication warnings from Pytest

### ISSUE THREAD

**Issue by ocampor** [computationalmodelling/nbval#139]
### Problem Description

I run some tests using nbval that raised the following warnings:
- PytestDeprecationWarning: direct construction of IPyNbFile has been deprecated, please use IPyNbFile.from_parent return IPyNbFile(path, parent)
- PytestDeprecationWarning: direct construction of IPyNbCell has been deprecated, please use IPyNbCell.from_parent
    cell, options)
-  DeprecationWarning: KernelManager._kernel_spec_manager_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead.
    def _kernel_spec_manager_changed(self):

### Environment Details:

```
Pytest Version: 5.4.1
Python Version: 3.7
Nbval Version: 0.9.5
```

I can probably create a PR to fix these issues if you want.

**Comment by vidartf** [comment]
> I can probably create a PR to fix these issues if you want.

PRs are always welcome :)

**Comment by GuillaumeFavelier** [comment]
I have the exact same behaviour with `pytest 5.4.3`

**Comment by bretttully** [comment]
Has anyone had a crack at this yet? It is now an error, rather than a warning:

```bash
============================= test session starts ==============================
platform linux -- Python 3.7.8, pytest-6.0.0, py-1.9.0, pluggy-0.13.1
rootdir: /root/***
plugins: cov-2.10.0, nbval-0.9.5
collected 0 items / 1 error
```

```bash
==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.7/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.7/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.7/site-packages/pluggy/manager.py:87: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.7/site-packages/nbval/plugin.py:115: in pytest_collect_file
    return IPyNbFile(path, parent)
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.7/site-packages/_pytest/nodes.py:95: in __call__
    warnings.warn(NODE_USE_FROM_PARENT.format(name=self.__name__), stacklevel=2)
E   pytest.PytestDeprecationWarning: Direct construction of IPyNbFile has been deprecated, please use IPyNbFile.from_parent.
E   See https://docs.pytest.org/en/stable/deprecations.html#node-construction-changed-to-node-from-parent for more details.
```

**Comment by omry** [comment]
Pytest 6.0.0 makes deprecation warnings treated as errors by default, with the intention of removing the workaround for it by 6.1.
https://docs.pytest.org/en/stable/changelog.html

Please fix this.

**Comment by omry** [comment]
Thanks for fixing.
As this is fixing a major issue, do you mind pushing up an updated package?

**Comment by takluyver** [comment]
@omry - Vidar already did, version 0.9.6.

**Comment by omry** [comment]
Oh great. I was fooled by pypi search which is still showing the old version.
![image](https://user-images.githubusercontent.com/376455/88950691-d539c500-d249-11ea-9dac-b668100c65b9.png)

**Comment by jli** [comment]
**tl;dr** The 3rd DeprecationWarning listed in the original issue is still happening. It's due to jupyter_client. They've fixed the issue, but haven't done a new release yet.

The DeprecationWarning isn't causing a failure for me, even on pytest-6.1.2. With a minimal Python 3 notebook (just 1 code cell: `1+1`), here's the output (lightly edited):

```shell
% pytest --nbval nbval_deprecation_test.ipynb
===================================== test session starts ======================================
platform darwin -- Python 3.9.0, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: .../my-repo, configfile: pytest.ini
plugins: nbval-0.9.6, cov-2.10.1
collected 1 item

nbval_deprecation_test.ipynb .                                                           [100%]

======================================= warnings summary =======================================
.../nbval_deprecation_test.ipynb::Cell 0
  .../lib/python3.9/site-packages/jupyter_client/manager.py:66: DeprecationWarning: KernelManager._kernel_spec_manager_changed is deprecated in traitlets 4.1: use @observe and @unobserve instead.
    def _kernel_spec_manager_changed(self):

-- Docs: https://docs.pytest.org/en/stable/warnings.html
================================= 1 passed, 1 warning in 1.28s =================================
```

It seems like this is a jupyter_client issue, fixed on 2020-11-11: https://github.com/jupyter/jupyter_client/pull/588

There hasn't been a jupyter_client release that includes the fix, though.
